tools/sklearn/pca.xml

```diff
@@ -0,0 +1,210 @@
+                        <option value="cosine">cosine</option>
+                        <option value="precomputed">precomputed</option>
+                    </param>
+                    <when value="poly">
+                        <param name="gamma" type="float" optional="true" label="Gamma Value" help="Kernel coefficient for rbf, poly and sigmoid kernels. Ignored by other kernels"/>
+                        <param name="degree" type="integer" value="3" label="Degree of the polynomial" help="Degree for poly kernels. Ignored by other kernels"/>
+                        <param name="coef0" type="float" value="1.0" label="Coef0" help="Independent term in poly and sigmoid kernels. Ignored by other kernels"/>
+                    </when>
+                    <when value="sigmoid">
+                        <param name="gamma" type="float" optional="true" label="Gamma Value" help="Kernel coefficient for rbf, poly and sigmoid kernels. Ignored by other kernels"/>
+                        <param name="coef0" type="float" value="1.0" label="Coef0" help="Independent term in poly and sigmoid kernels. Ignored by other kernels"/>
+                    </when>
+                    <when value="rbf">
+                        <param name="gamma" type="float" optional="true" label="Gamma Value" help="Kernel coefficient for rbf, poly and sigmoid kernels. Ignored by other kernels"/>
+                    </when>
+                </conditional>
+                <conditional name="select_solver_type">
+                    <param name="eigen_solver_opts" type="select" label="Eigen Solver">
+                        <option value="auto" selected="true">auto</option>
+                        <option value="dense">dense</option>
+                        <option value="arpack">arpack</option>
+                    </param>
+                    <when value="arpack">
+                        <param name="tolerance" type="float" value="0.0" label="Tolerance" help="Convergence tolerance for arpack. If 0, optimal value will be chosen by arpack"/>
+                        <param name="max_iter" type="integer" optional="true" label="Maximum Iterations" help="Maximum number of iterations for arpack"/>
+                    </when>
+                </conditional>
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data format="tabular" name="outfile"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="infile" value="pca_input.dat" ftype="tabular"/>
+            <param name="select_cols_opts" value="include"  />
+            <param name="column_indices" value="0 1 3 5 7 4" />
+            <param name="number" value="5"/>
+            <param name="select_pca_opts" value="classical"/>
+            <param name="svd_solver_opts" value="arpack"/>
+            <param name="tolerance" value="0.4"/>
+            <output name="outfile" ftype='tabular' compare="sim_size" delta="1" file="pca_classical_output.dat"/>
```

Comment:
for each of these tests, can you remove `compare="sim_size"`? It should be possible to compare the outputs exactly